Predictors/rnn.py

```diff
@@ -0,0 +1,172 @@
+from math import floor
+
+import numpy as np
+import tensorflow as tf
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+
+class RNNClassifier(BaseEstimator, ClassifierMixin):
+    """
+    Generic RNN - can use ordinary RNN, LSTM or GRU
+    
+    Internal state is fed into a fully connected logistic regression layer and then softmax is applied to achieve classification
+       Basically the logistic regression layer provides a probability for each class.
+    
+    Based upon "Training a Sequence Classifier" on page 587 of "Hands on Machine Learning with Scikit-Learn and TensorFlow" by Aurelien Geron
+    """
+
+    def __init__(self, CellType=tf.contrib.rnn.GRUCell, n_neurons=100, learning_rate=0.001, n_outputs=4,
+                 n_out_neurons=10, n_inputs=50, n_steps=5, n_epochs=100, batch_size=100, **cell_args):
+        """
+        n_inputs = word encoding size
+        n_steps = amount of recursion e.g. window size
+        n_neurons = number of neurons in RNN
+        n_out_neurons = number of neurons in the fully connected output layer (= the number of classes)
+        n_outputs = number of outputs
+        """
+        self.CellType = CellType
```

Comment:
instance variables should be snake case (I don't see this being required in the inherited classes), also I notice CellType doesn't feature in the docstring above

Reply:
CellType is a class rather than an instance. Is there a way to make that distinction or should it just be named like everything else?

Reply:
the type of the variable shouldn't really define which naming convention it uses. CapWords is reserved for actual names of classes.